### PR TITLE
Remove dependency on github.com/pkg/errors

### PR DESCRIPTION
**Pull request by pacoxu**
Fixes https://github.com/kubernetes/system-validators/issues/33

### CODE REVIEW

File: validators/cgroup_validator_linux.go [hunk 1]
```diff
@@ -55,32 +55,32 @@ func (c *CgroupsValidator) Validate(spec SysSpec) (warns, errs []error) {
 	var st unix.Statfs_t
 	var err error
 	if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
-		return nil, []error{errors.Wrap(err, "cannot statfs the cgroupv2 root")}
+		return nil, []error{fmt.Errorf("cannot statfs the cgroupv2 root: %v", err)}
```
Comment (SataQiu) [inline validators/cgroup_validator_linux.go:58]: How about using `: %w` here?
For example: https://go.dev/play/p/P9R4uGYZaqa

File: validators/package_validator_linux.go [hunk 2]
```diff
@@ -245,9 +244,10 @@ func extractUpstreamVersion(version string) string {
 
 // toSemVerRange converts the input to a semantic version range.
 // E.g., ">=1.0"             -> ">=1.0.x"
-//       ">=1"               -> ">=1.x"
-//       ">=1 <=2.3"         -> ">=1.x <=2.3.x"
-//       ">1 || >3.1.0 !4.2" -> ">1.x || >3.1.0 !4.2.x"
+//
```
Comment (SataQiu) [inline validators/package_validator_linux.go:247]: Emm... the previous format seems better.

Comment (pacoxu) [reply validators/package_validator_linux.go]: The `update-gofmt.sh` changed this. Or there will be a format verification failure.

I updated it to add ` - ` before it. Is it better now?

File: validators/cgroup_validator_linux.go (hunk 1 above)
Comment (pacoxu) [reply validators/cgroup_validator_linux.go]: Updated.

File: validators/cgroup_validator_linux.go [hunk 3]
```diff
@@ -55,32 +55,32 @@ func (c *CgroupsValidator) Validate(spec SysSpec) (warns, errs []error) {
...
-			return nil, []error{errors.Wrap(err, "failed to get cgroup v2 subsystems")}
+			return nil, []error{fmt.Errorf("failed to get cgroup v2 subsystems: %w", err)}
 		}
 		requiredCgroupSpec = spec.CgroupsV2
 		optionalCgroupSpec = spec.CgroupsV2Optional
 	} else {
 		subsystems, err = c.getCgroupV1Subsystems()
 		if err != nil {
-			return nil, []error{errors.Wrap(err, "failed to get cgroup v1 subsystems")}
+			return nil, []error{fmt.Errorf("failed to get cgroup v1 subsystems: %vs", err)}
```
Comment (neolit123) [inline validators/cgroup_validator_linux.go:73]:
```suggestion
			return nil, []error{fmt.Errorf("failed to get cgroup v1 subsystems: %w", err)}
```

**Review by neolit123 — COMMENTED** [review]
/approve

File: validators/package_validator_linux.go [hunk 4]
```diff
@@ -46,7 +45,7 @@ func newPackageManager() (packageManager, error) {
 	if m, ok := newDPKG(); ok {
 		return m, nil
 	}
-	return nil, errors.New("failed to find package manager")
+	return nil, fmt.Errorf("failed to find package manager")
```
[the remark this answers is not in the document]
Comment (pacoxu) [reply validators/package_validator_linux.go:48]: - fmt.Errorf("xxx")
- errors.New("xxx")

Either works and `errors.New` is a little better in my mind.

[the remark this answers is not in the document]
Comment (pacoxu) [reply validators/package_validator_linux.go:48]: image

Two places were using `fmt.Errorf("xxx")` in the master branch.
If we decided to use `errors.New()` instead, I will update them together.

[the remark this answers is not in the document]
Comment (neolit123) [reply validators/package_validator_linux.go:48]: yes, it's an additional import for errors.New, but it's the more accurate way. would have been better if stdlib had errors.Errorf too.

[the remark this answers is not in the document]
Comment (pacoxu) [reply validators/package_validator_linux.go]: Updated.

**Review by SataQiu — COMMENTED** [review]
/approve
/lgtm

**Comment by pacoxu** [comment]
Only this PR was merged after the last release v1.8.0. I am not sure if a new release for v1.27 is needed. Either would be OK.
Probably v1.8.1.

**Pull request merged**